distributed/tests/test_cancelled_state.py

```diff
@@ -164,8 +164,8 @@ def blockable_compute(x, lock):
         await block_compute.acquire()
 
         # Close in scheduler to ensure we transition and reschedule task properly
-        await s.remove_worker(a.address, stimulus_id="test")
-        await wait_for_state(fut1.key, "resumed", b)
+        s.close_worker(worker=a.address)
+        await wait_for_state(fut1.key, "resumed", b, interval=0)
```

Comment:
I think you have a race condition between `c.submit` and `await block_compute.acquire()`; they should be swapped around. At that point you would not need to bother with the interval.

Reply:
rewrote the test slightly. Works very reliably on my machine